src/Npgsql/NpgsqlCommand.cs

```diff
@@ -891,7 +891,7 @@ internal void ProcessRawQuery(SqlQueryParser? parser, bool standardConformingStr
                 break;
 
             default:
-                ThrowHelper.ThrowArgumentOutOfRangeException(nameof(PlaceholderType), $"Unknown {nameof(PlaceholderType)} value: {Parameters.PlaceholderType}");
+                ThrowHelper.ThrowArgumentOutOfRangeException(nameof(PlaceholderType), $"Unknown {nameof(PlaceholderType)} value: {{0}}", Parameters.PlaceholderType);
```

Comment:
So just to be sure, these changes actually make the method get inlined? Because if not, I'm not sure it makes sense to do it everywhere systematically etc

Reply:
I don't think `ProcessRawQuery` can ever get inlined (not like we even want that), but that way we can get rid of string interpolation and potentially a few operations. I do not have any strong opinion about leaving it as is or not.

Reply:
Right, though e.g. here the string interpolation (and the rest) only occur in case of an exception (and in this case, in an impossible assertion exception) - so there's really no need to optimize it.

I think this kind of micro-optimization makes sense where it helps, e.g. because it allows inlining which otherwise wouldn't be possible. For other cases, I'd personally rather keep the code more readable, though I could see a case for doing it the same way across the entire codebase (i.e. for consistency).

Reply:
> Right, though e.g. here the string interpolation (and the rest) only occur in case of an exception (and in this case, in an impossible assertion exception) - so there's really no need to optimize it.

That's a comparison with and without this specific change (on the left is without): https://www.diffchecker.com/3HJwh4D5/

Just by having string interpolation we're incuring the cost of 5 additional operations per every `ProcessRawQuery` call:

```asm
vxorps    xmm4,xmm4,xmm4
vmovdqa   xmmword ptr [rsp+40],xmm4
vmovdqa   xmmword ptr [rsp+50],xmm4
xor       eax,eax
mov       [rsp+60],rax
```

It is a micro-optimizations and I'm okay leaving it as is, just wanted to point out that by optimizing failure paths we might also optimize good paths.